api/v1alpha1/kvmmachine_types.go

```diff
@@ -0,0 +1,131 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha4"
+	"sigs.k8s.io/cluster-api/errors"
+)
+
+// KVMMachineSpec defines the desired state of KVMMachine
+type KVMMachineSpec struct {
+	// ProviderID is the unique identifier of the machine assigned by the provider.
+	ProviderID *string `json:"providerID,omitempty"`
+
+	// Resources defines the compute and storage requirements of the machine.
+	Resources Resources `json:"resources,omitempty"`
+
+	// HostVolumes defines the host volumes (PV) to be mounted into the workload cluster machine.
+	// +optional
+	// +nullable
+	HostVolumes []HostVolume `json:"hostVolumes,omitempty"`
```

Comment:
We have a chance to make this interface nicer--could potentially support all Kubernetes volume types instead of this rigid structure if desired.